### PR TITLE
Include Salvo AMIs and Snapshots in the cleanup.

### PR DESCRIPTION
**Pull request by mum4k**
Also added unit tests for the new code.

### CODE REVIEW

File: instances/azp-cleanup-snapshots/ami_filter.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// Functions that assist when filtering through AMIs.
+
+/**
+ * Determines whether the AMI and its snapshots should be cleaned by this
+ * Lambda based on the tags it has.
+ * @param tags - an array of all the tags attached to the AMI.
+ * @returns {boolean} - true if the AMI is in scope and should be cleaned up.
+ * False otherwise. Note that the Lambda only cleans up the AMI if unused.
+**/
+exports.isAmiInScopeFromTags = function (tags) {
```
Comment (phlax) [inline instances/azp-cleanup-snapshots/ami_filter.js]: any reason why this doesnt use arrow fun ?

File: instances/azp-cleanup-snapshots/test/ami_filter.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+// the argument project.
+function createAmiTagsWithProject(project) {
+    const tag = new Object();
+    tag.Key = 'Project';
+    tag.Value = project;
+
+    return new Array(tag);
+}
+
+describe('isAmiInScopeFromTags unit test)', function () {
```
Comment (phlax) [inline instances/azp-cleanup-snapshots/test/ami_filter.test.js]: same i guess with these

File: instances/azp-cleanup-snapshots/ami_filter.js (hunk 1 above)
Comment (mum4k) [reply instances/azp-cleanup-snapshots/ami_filter.js]: Changed to arrow funcs. Thank you.

File: instances/azp-cleanup-snapshots/test/ami_filter.test.js (hunk 2 above)
Comment (mum4k) [reply instances/azp-cleanup-snapshots/test/ami_filter.test.js]: Done.

**Review by fei-deng — APPROVED** [review]
LGTM.

**Review by phlax — APPROVED** [review]
lgtm, thanks @mum4k

**Comment by mum4k** [comment]
I am going to push this out and confirm the lambda still works before we merge it into main.

**Review by gyohuangxin — APPROVED** [review]
Thanks!

**Comment by mum4k** [comment]
I have applied this change via Terraform and will observe for some time.

**Comment by mum4k** [comment]
The Lambda still works, so going to merge this.

I wasn't able to confirm that this also cleans up Salvo correctly yet, because we don't have multiple resources up yet. I will confirm later when we deploy more Salvo resources.

**Pull request merged**